process/smartContract/sovereignSCProcess.go

```diff
@@ -0,0 +1,102 @@
+package smartContract
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/smartContractResult"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/state"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type sovereignSCProcessor struct {
+	*scProcessor
+}
+
+// TODO: use scrProcessorV2 when feat/vm1.5 is merged into feat/chain-sdk-go
+
+// NewSovereignSCRProcessor creates a sovereign scr processor
+func NewSovereignSCRProcessor(scrProc *scProcessor) (*sovereignSCProcessor, error) {
+	if check.IfNil(scrProc) {
+		return nil, process.ErrNilSmartContractResultProcessor
+	}
+
+	sovereignSCProc := &sovereignSCProcessor{
+		scProcessor: scrProc,
+	}
+
+	sovereignSCProc.getAccountFromAddressFunc = sovereignSCProc.getAccountFromAddress
+
+	return sovereignSCProc, nil
+}
+
+// ProcessSmartContractResult updates the account state from the smart contract result
+func (sc *sovereignSCProcessor) ProcessSmartContractResult(scr *smartContractResult.SmartContractResult) (vmcommon.ReturnCode, error) {
+	if check.IfNil(scr) {
+		return 0, process.ErrNilSmartContractResult
+	}
+
+	log.Trace("sovereignSCProcessor.ProcessSmartContractResult()", "sender", scr.GetSndAddr(), "receiver", scr.GetRcvAddr(), "data", string(scr.GetData()))
+
+	var err error
+	returnCode := vmcommon.UserError
+	if !bytes.Equal(scr.SndAddr, core.ESDTSCAddress) {
+		return returnCode, fmt.Errorf("%w, expected ESDTSCAddress", errInvalidSenderAddress)
+	}
+
+	scrData, err := sc.checkSCRBeforeProcessing(scr)
+	if err != nil {
+		return returnCode, err
+	}
+
+	txType, _ := sc.txTypeHandler.ComputeTransactionType(scr)
+	switch txType {
+	case process.BuiltInFunctionCall:
+		err = sc.checkBuiltInFuncCall(string(scr.Data))
+		if err != nil {
+			return returnCode, err
+		}
+
+		return sc.ExecuteBuiltInFunction(scr, nil, scrData.destination) //nil sender account
+	default:
+		err = process.ErrWrongTransaction
+	}
+
+	return returnCode, sc.ProcessIfError(scrData.sender, scrData.hash, scr, err.Error(), scr.ReturnMessage, scrData.snapshot, 0)
+}
+
+func (sc *sovereignSCProcessor) checkBuiltInFuncCall(scrData string) error {
+	function, _, err := sc.argsParser.ParseCallData(scrData)
+	if err != nil {
+		return err
+	}
+
+	if function != core.BuiltInFunctionMultiESDTNFTTransfer {
+		return fmt.Errorf("%w, expected %s", errInvalidBuiltInFunctionCall, core.BuiltInFunctionMultiESDTNFTTransfer)
+	}
+
+	return nil
+}
+
+func (sc *sovereignSCProcessor) getAccountFromAddress(address []byte) (state.UserAccountHandler, error) {
+	addrShard := sc.shardCoordinator.ComputeId(address)
+	// TODO: This will not work for now, since coordinator does not know how to return correct sovereign shard ID
```

Comment:
in case of sovereignSCProcessor there is not need to compare addrShard with core.SovereignChainShardID. And I do not see the need to change the shardCoordinator into knowing about sovereign shard. Maybe we could have another coordinator - sovereignCoordinator let's say - which returns mainChainID or SovereignChainID (or later Sovereign1, Sovereign2 and so forth.).

Reply:
> in case of sovereignSCProcessor there is not need to compare addrShard with core.SovereignChainShardID. And I do not see the need to change the shardCoordinator into knowing about sovereign shard.

Actually, right, we do not need this. I think we can safely delete it completely

> Maybe we could have another coordinator - sovereignCoordinator let's say - which returns mainChainID or SovereignChainID (or later Sovereign1, Sovereign2 and so forth.).

yes, there is already a task for this